weather.py

```diff
@@ -0,0 +1,306 @@
+# Imports written in alphabetical order
+from os import path
+import sys
+from termcolor import colored
+import zipfile
+
+
+class Weatherman:  # Weatherman Application in one class
+
+    file_pre = "Murree_weather_"  # The part of the file name that is same for all files
+
+    # Dictionaries as class instances to store calculation results
+    e_result = dict()
+    a_result = dict()
+    c_result = dict()
+
+    # Report Number to print
+    report_num = 1
+
+    # Initialize Lists in these indexes of dictionary
+    c_result["max_temp"] = []
+    c_result["min_temp"] = []
```

Comment:
Class variables are shared between all class instances which means if you will create two instances they will access the same class variables and if one instance will change the values it will get changed for the other instance as well. These variables are not global states so it's better to use instance variables instead of class variables. Also e, a and c don't make any sense if you can come up with better names it will be great.

Reply:
change made as suggested.